wpiutil/src/main/native/cpp/Synchronization.cpp

```diff
@@ -17,7 +17,7 @@
 
 using namespace wpi;
 
-static std::atomic_bool gShutdown{false};
+static std::atomic_int gActive{0};
```

Comment:
Any reason not to use a semaphore here to refcount?  That would be simpler than atomics.  It isn't async safe, but the rest of this code looks complicated enough that feels un-necessary.

Or maybe the real request is, "can you add a couple of comments capturing why this is needed"?

Reply:
Added a writeup.